v4/export/dump.go

```diff
@@ -851,6 +851,18 @@ func extractTiDBRowIDFromDecodedKey(indexField, key string) (string, error) {
 	return "", errors.Errorf("decoded key %s doesn't have %s field", key, indexField)
 }
 
+func getListTableTypeByConf(conf *Config) listTableType {
+	listType := listTableByInfoSchema
+	if conf.Consistency == consistencyTypeLock {
+		// for consistency lock, we need to build the tables to dump as soon as possible
+		listType = listTableByShowTableStatus
+	}
+	if conf.Consistency == consistencyTypeFlush && matchMysqlBugversion(conf.ServerInfo) {
+		listType = listTableByShowFullTables
+	}
```

Comment:
This doesn't seem to match the logic before.
```suggestion
	listTableType := listTableByShowTableStatus
	if conf.Consistency == consistencyTypeLock {
		// for consistency lock, we need to build the tables to dump as soon as possible
		listTableType = listTableByInfoSchema
	} else if matchMysqlBugversion(conf) {
		listTableType = listTableByShowFullTables
	}
```